.github/workflows/test.yml

```diff
@@ -85,6 +87,17 @@ jobs:
           compiler: ${{ matrix.toolchain.compiler }}
           version: ${{ matrix.toolchain.version }}
       
+      - name: Test MKL libraries
+        # we could also exclude 2021.5 for all macos versions here
+        # this needs refactoring at some point - should the install
+        # fail if no mkl version exists?
```

Comment:
I think it makes sense to fail, given we aim to guarantee full support barring a few exceptions. In general I think we should fail if we can't satisfy the user's selected configuration.